WeakAuras/BuffTrigger2.lua

```diff
@@ -1884,14 +1884,14 @@ local function createScanFunc(trigger)
       end
     ]]
   end
-
-  if use_debuffClass and trigger.debuffClass then
+  if use_debuffClass then
     local ret2 = [[
-      if matchData.debuffClass ~= %q then
+      local tDebuffClass = %s;
+      if not tDebuffClass[matchData.debuffClass] then
         return false
       end
     ]]
-    ret = ret .. ret2:format(trigger.debuffClass)
+    ret = ret .. ret2:format(trigger.debuffClass and type(trigger.debuffClass) == "table" and WeakAuras.SerializeTable(trigger.debuffClass) or "{}")
```

Comment:
One small note: this will cause a small memory leak, which someone is bound to complain about. I don't personally care that much unless it causes a serious issue for somebody, but somebody else might care.

Reply:
I'm not good for spotting this, do you have suggestions ?

Maybe instead of using WeakAuras.SerializeTable(trigger.debuffClass) to make a new table it could use string ? That doesn't sound very efficient to me but maybe it is better to handle a string like with the following example  ?

```Lua
local t = {
  truc = true,
  chose = true,
  bidule = true
}
local matchData_debuffClass = "bidule"

function table2str(t)
  local keyset={}
  local n=0
  for k,v in pairs(t) do
    n=n+1
    keyset[n]=k
  end
  return table.concat(keyset,"#")
end
local tDebuffClass = table2str(t)
print(string.match(tDebuffClass , matchData_debuffClass))
```

Reply:
That doesn't matter imho, that code only runs once per trigger construction. So typically once.